app/main.py

```diff
@@ -1,8 +1,26 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    persons = []
+    for person in people:
+        obj = Person(person["name"], person["age"])
+        partner = "wife" if "wife" in person else "husband"
+        setattr(obj, partner, person[partner])
```

Comment:
The `setattr` function is setting the `wife` or `husband` attribute directly from the dictionary, which does not ensure that these attributes are references to other `Person` instances. You should initially set these attributes to `None` or skip setting them here, and then set them to reference the correct `Person` instance in the second loop.